control/gps.py

```diff
@@ -37,9 +66,17 @@ def __repr__(self):
     def read(self):
         """Returns a GpsReading object with the values supplied"""
         msg = None
-        # TODO: Add try except chain to catch any dirty GPS output exceptions
-        # TODO: Add timeout for reads using signal library
-        while not isinstance(msg, pynmea2.GGA):
-            msg = pynmea2.parse(self.ser.readline())
+        tries = 4
```

Comment:
I'd name this `max_tries` instead of `tries`, just to illustrate it's a limit and not the current number of tries, but I am being crazy ocd there. I think it is clear enough from the context. If you do make other changes in here for some reason consider that otherwise just leave it.

Reply:
I can update this when we add the next helper function for converting GPS to meters if we don't want a commit just for this. Otherwise I can change it real quick.

Reply:
Fixed